File: backend/questions/template_question_1/java/Grader.java
import java.io.*;
import java.util.*;

public class Grader {
  public static void main(String[] args) {
    try {
      File input = new File("../input");
      File output = new File("../output");
      Scanner inputReader = new Scanner(input);
      Scanner outputReader = new Scanner(output);
      int total = 0, correct = 0;
      Solution solution = new Solution();
      while (inputReader.hasNextLine()) {
        total++;
        String s1 = inputReader.nextLine();
        String s2 = inputReader.nextLine();
        boolean ans  = outputReader.nextLine().equals("true");
        boolean result = solution.solve(s1, s2);
        if (result == ans) correct++;
      }
      inputReader.close();
      outputReader.close();
      System.out.printf("%d/%d test cases passed!\n", correct, total);
    } catch (Exception e) {
      e.printStackTrace();
    }
  }
}

File: backend/questions/template_question_1/javascript/grader.js
const fs = require("fs");
const solution  = require("./user_solution");

fs.readFile("../input", (err, input) => {
    if (err) throw err;

    fs.readFile("../output", (err, output) => {
        if (err) throw err;

        const inputs = input.toString().split('\n');
        const outputs = output.toString().split('\n');

        var total = outputs.length;
        var correct = 0;

        for (var i = 0; i < total; i++) {
            const result = solution.solve(inputs[i * 2], inputs[i * 2 + 1]);
            const ans = outputs[i] === "true";
            if (result === ans) correct++;
        }

        console.log("%d/%d test cases passed!", correct, total);
    })
});